doc/devel/contributing.rst

```diff
@@ -0,0 +1,414 @@
+.. _contributing:
+
+============
+Contributing
+============
+
+This project is a community effort, and everyone is welcome to
+contribute.
+
+The project is hosted on https://github.com/matplotlib/matplotlib
+
+Submitting a bug report
+=======================
+
+If you find a bug in the code or documentation, do not hesitate to submit a
+ticket to the
+`Bug Tracker <https://github.com/matplotlib/matplotlib/issues>`_. You are
+also welcome to post feature requests or pull requests.
+
+
+Retrieving and installing the latest version of the code
+========================================================
+
+
+We use `Git <https://git-scm.com/>`_ for version control and
+`GitHub <https://github.com/>`_ for hosting our main repository.
+
+You can check out the latest sources with the command::
+
+    git clone git@github.com:matplotlib/matplotlib.git
+
+
+After obtaining a local copy of the matplotlib source code (:ref:`set-up-fork`),
+navigate to the matplotlib directory and run the following in the shell::
+    
+    python setup.py develop
+
+or::
+  
+   pip install -v -e .
+
+
+This installs matplotlib for development (i.e., builds everything and places the
+symbolic links back to the source code).
+
+.. warning::
+
+   If you already have a version of matplotlib installed, you will need to
+   uninstall it.
+
+
+.. note::
+  
+    If you decide to do install with ``python setup.py develop`` or ``pip
+    install -v -e``, you will have to rerun::
+   
+      python setup.py build
+
+    every time the source code of a compiled extension is changed (for
+    instance when switching branches or pulling changes from upstream).
+
+
+
+You can then run the tests to check your work environment is set up properly::
+
+  python tests.py
+
+
+.. _nose: https://nose.readthedocs.io/en/latest/
+.. _pep8: https://pep8.readthedocs.io/en/latest/
+
+.. note::
+    
+  **Additional dependencies for testing**: nose_ (version 1.0 or later), `mock
+  <https://docs.python.org/dev/library/unittest.mock.html>`_ (if python < 3.3), `Ghostscript
+  <http://www.ghostscript.com/>`_, `Inkscape <http://inkscape.org>`_
+
+.. note:: To make sure the tests run locally:
+
+      * Copy setup.cfg.template to setup.cfg
```

Comment:
I am not sure what you are referring too.
I actually never had to do this to have the tests run on my computer.